common/bootfirmware_info.c

```diff
@@ -14,20 +14,28 @@ int __weak get_boot_firmware_info(void)
 {
 	char *version;
 	int node;
+	int ret = 0;
 
 	node = fdt_node_offset_by_compatible(gd->fdt_blob, -1, COMPATIBLE);
 	if (node < 0) {
 		printf("Can't find node with compatible = \"" COMPATIBLE
 		       "\"\n");
-		return -FDT_ERR_NOTFOUND;
+		ret = -FDT_ERR_NOTFOUND;
+		goto out:
 	}
 
 	version = fdt_getprop(gd->fdt_blob, node, "bootfirmware-version", NULL);
 	if (version) {
 		printf("Boot firmware version: %s\n", version);
 		env_set("dt_bootfirmware_version", version);
 	} else {
-		return -FDT_ERR_NOTFOUND;
+		ret = -FDT_ERR_NOTFOUND;
+	}
+
+out:
+	/* we should stop booting, if boot firmware version is not found */
+	if (CONFIG_IS_ENABLED(BOOTFIRMWARE_INFO_STRICT)) {
+		return ret;
```

Comment:
What is the outcome of returning error at this stage?

Reply:
U-Boot will print `initcall sequence <address> failed at call <address> (err=<error>)` and hang.

Reply:
Should we add a print message here explaining why the failure then?